www_src/components/color-spectrum/color-spectrum.jsx

```diff
@@ -1,20 +1,116 @@
 var React = require('react');
+var Slider = require('../../components/range/range.jsx');
+var Color = require('color');
 
-// Doesn't do much yet, but renders the hue and saturation spectrum
+var HSB = React.createClass({
+  getBackgroundColor: function () {
+    var color = this.props.color.clone();
+    color.hsv(color.hue(), 100, 100);
+    return color.rgbaString();
+  },
+  getSBPosition: function () {
+    var hsv = this.props.color.hsv();
+    return {
+      top: `${100 - hsv.v}%`,
+      left: `${hsv.s}%`
+    };
+  },
+  changeHue: function (e) {
+    var color = this.props.color;
+    color.hue(e.target.value);
+    this.props.onChange(color);
+  },
+  changeSB: function (e) {
+    var color = this.props.color;
 
-var ColorSpectrum = React.createClass({
+    var x = e.touches[0].clientX - this.bounding.left;
+    var y = e.touches[0].clientY - this.bounding.top;
+
+    color.hsv(color.hue(), x / this.bounding.width * 100, 100 - y / this.bounding.height * 100);
+    this.props.onChange(color);
+  },
+  onTouchStart: function(e) {
+    e.preventDefault();
+    this.bounding = this.refs.sl.getDOMNode().getBoundingClientRect();
+    this.changeSB(e);
+  },
+  onTouchMove: function (e) {
+    this.changeSB(e);
+  },
   render: function () {
+
     return (<div className="spectrum-container">
       <div className="spectrum-left">
-        <div className="saturation" style={{backgroundColor: this.props.value}}>
+        <div ref="sl" className="saturation" style={{backgroundColor: this.getBackgroundColor()}} onTouchStart={this.onTouchStart} onTouchMove={this.onTouchMove}>
```

Comment:
strongly advice to just be verbose here. "sl" is not a very meaningful word, do have have a better candidate?